### PR TITLE
Add option for cinc instead of chef

### PR DESCRIPTION
**Pull request by timhassett**
Add option for cinc instead of chef
Allow specifying EBS config

### CODE REVIEW

File: vars/packer.groovy [hunk 1]
```diff
@@ -122,6 +124,16 @@ def call(body) {
...
+  if (config.ebsDeviceName) {
+    ebsDeviceName = config.ebsDeviceName
+  }
+  if (config.ebsVolumeSize) {
+    ebsVolumeSize = config.ebsVolumeSize
+  }
+  if ((ebsDeviceName && !ebsVolumeSize) || (!ebsDeviceName && ebsVolumeSize)) {
+    throw new GroovyRuntimeException("Supply both ebs Volumes Size and DeviceName")
+  }
+
```
Comment (Guslington) [inline vars/packer.groovy:136]: you could simplify this logic to as `config.ebsDeviceName` will return null if not set

```groovy
if ((config.ebsDeviceName && !config.ebsVolumeSize) || (!config.ebsDeviceName && config.ebsVolumeSize)) {
    throw new GroovyRuntimeException("Supply both ebs Volumes Size and DeviceName to set a volume in the packer template")
}

if (config.ebsDeviceName && config.ebsVolumeSize) {
    ptb.addBlockDevice(config.ebsDeviceName, config.ebsVolumeSize)
}

File: src/com/base2/ciinabox/PackerTemplateBuilder.groovy [hunk 2]
```diff
@@ -78,9 +78,14 @@ class PackerTemplateBuilder implements Serializable {
     if (json) {
       chefProvisioner.json = json
     }
-
+    def use_cinc = false
```
Comment (Guslington) [inline src/com/base2/ciinabox/PackerTemplateBuilder.groovy:81]: i would create a separate parameter to set packer to use cinc to make it more obvious to configure than prefixing the version with `cinc-`

**Review by Guslington — CHANGES_REQUESTED** [review]

File: vars/packer.groovy (hunk 1 above)
Comment (timhassett) [reply vars/packer.groovy:130]: Change applied

File: src/com/base2/ciinabox/PackerTemplateBuilder.groovy (hunk 2 above)
Comment (timhassett) [reply src/com/base2/ciinabox/PackerTemplateBuilder.groovy]: Created `useCinc` and `cincVersion` params for packer function

File: vars/packer.groovy [hunk 3]
```diff
@@ -158,7 +167,11 @@ def call(body) {
       config.get('cookbookS3Bucket'),
       config.get('cookbookS3Region', region),
       config.get('cookbookS3Path'))
-    ptb.addChefSoloProvisioner(config.runList,config.get('chefJSON'),config.get('chefVersion'))
+    if (config.get('useCinc', false)) {
```
Comment (Guslington) [inline vars/packer.groovy:170]: no need for the if statement, you could just pass the value of `config.get('useCinc', false)` in to the method as the method handles the logic

```groovy
ptb.addChefSoloProvisioner(config.runList,config.get('chefJSON'),config.get('cincVersion'), config.get('useCinc', false))

**Review by Guslington — CHANGES_REQUESTED** [review]

**Review by Guslington — APPROVED** [review]

**Pull request merged**